Single Page Map

drewjj opened the pull request:
- Combines the following pages into a single page map:
  - RSU map
  - Heatmap
  - BSM map
  - WZDx map
![image](https://user-images.githubusercontent.com/31270488/231120443-9561b158-60e4-4ac1-b927-176226828820.png)
- Togglable map legend to control which layers are actively being displayed
![image](https://user-images.githubusercontent.com/31270488/231120776-1d0e311a-04a3-4490-8fb2-3af790c1c7cf.png)
- Remove unnecessary pages from the project and upper tab bar
- Heatmap logic code has been pulled into the redux RSU slice

Note: Currently Map.js is a fairly large file. This should be broken up into separate function files to make things a little more maintainable.

drewjj committed:
Include devops pipelines

drewjj committed:
vscode config

drewjj committed:
Added toggle-able legend with mapbox layers

drewjj committed:
Single page map

drewjj committed:
Remove unused pages from project and site

drewjj committed:
Add SCMS and online status toggle button for more control

drewjj committed:
Removed CDOT specification from many of the application descriptions

payneBrandon commented on .vscode/launch.json:
```diff
@@ -0,0 +1,28 @@
+            },
+            "args": ["run"],
+            "envFile": "${workspaceRoot}/api/.env",
+            "preLaunchTask": "build-python-flask"
```
Should this be `build-python-api` to match the tasks.json title for it?

drewjj replied:
Addressed

payneBrandon commented on webapp/src/slices/rsuSlice.js:
```diff
@@ -1,541 +1,613 @@
+                    endDate: currentState.rsu.value.endDate,
+                })
+            ),
+            // dispatch(_getSsmSrmData({ token })),
```
Did we mean to leave this commented?

drewjj replied:
Good catch. I noticed this function was causing the initial load for the webapp to take much longer than it should be. I have now moved this to be called directly instead of a part of the startup getRsuData function. It only runs when a MAP is being visualized instead of before.

drewjj committed:
Update docker-compose to include comment about BUILD_PATH env var

payneBrandon reviewed (changes requested):
A couple comments, also troubleshooting on my end the issue we talked about in person.

drewjj committed:
Address pull request comments and adjust popup behavior when layer is hidden

drewjj commented:
I addressed a comment from Michael about how clicking on a RSU and hiding the RSU layer will result in the RSU pop-up and RSU configuration side menu as still being visible. This is a little strange so I have now addressed this. When hiding the RSU layer, the currently selected RSU will be unselected. The same logic has been applied to WZDx.

payneBrandon reviewed (approved):
Looks good to me!
